### ISSUE TITLE
Several streaming APIs broken on large files

### ISSUE THREAD

**Issue by controversial** [oven-sh/bun#1940]
### What version of Bun is running?

0.5.1 / 0.5.2

### What platform is your computer?

Darwin 22.3.0 arm64 arm

### What steps can reproduce the bug?

Attempt to use a streaming API (`fs.createReadStream`, `Bun.file(...).stream()`) to read a large file (around a few megabytes).

### What is the expected behavior?

The entirety of the file should be streamed.

### What do you see instead?

The stream stops before the end of the file.

### Additional information

This is a follow-up to #1939 with a few additional bugs we found in [discord](https://discordapp.com/channels/876711213126520882/876711213126520885/1069553712097407037).

Full reproduction:
```ts
import { fileURLToPath } from 'url';
import fs from 'fs';
import { Readable } from 'stream';

const path = '/path/to/my/large/file';


// 1. Ground truth

const str = fs.readFileSync(path, 'utf8');
console.log(`True length: ${str.length.toLocaleString()} (${str.split('\n').length.toLocaleString()} lines)`);


// 2. Bun async iterator

const file2 = Bun.file(path);
let length2 = 0;
let lines2 = 1;
for await (const chunk of file2.stream()) {
  length2 += chunk.length;
  chunk.forEach((c) => { if (c === '\n'.charCodeAt(0)) lines2++ });
}
console.log(`Bun async iterator length: ${length2.toLocaleString()} (${lines2.toLocaleString()} lines)`);


// 3. fs.createReadStream

const stream3 = fs.createReadStream(path);
let length3 = 0;
let lines3 = 1;
await new Promise<void>((resolve) => {
  stream3.on('data', (chunk) => {
    length3 += chunk.length;
    [...chunk.toString()].forEach((c) => { if (c === '\n') lines3++ });
  });
  stream3.on('end', () => resolve());
});
console.log(`fs.createReadStream length: ${length3.toLocaleString()} (${lines3.toLocaleString()} lines)`);


// 4. Readable.fromWeb

const stream4 = Readable.fromWeb(Bun.file(path).stream());
let length4 = 0;
let lines4 = 1;
await new Promise<void>((resolve) => {
  stream4.on('data', (chunk) => {
    length4 += chunk.length;
    [...chunk.toString()].forEach((c) => { if (c === '\n') lines4++ });
  });
  stream4.on('end', () => resolve());
});
console.log(`Readable.fromWeb length: ${length4.toLocaleString()} (${lines4.toLocaleString()} lines)`);
```

Output for my example file (4547215 bytes / 8903 lines):
```
True length: 4,647,215 (8,903 lines)
Bun async iterator length: 4,194,304 (8,032 lines)
fs.createReadStream length: 2,359,296 (4,462 lines)
Readable.fromWeb length: 2,326,528 (4,399 lines)
```

**Comment by Jarred-Sumner** [comment]
The streaming APIs for files share the same internal data structure (which is different from blob and HTTP responses)

**Comment by Jarred-Sumner** [comment]
This was fixed in #2029
